libs/feature/record/src/lib/map-view/map-view.component.ts

```diff
@@ -135,15 +141,68 @@ export class MapViewComponent implements AfterViewInit {
         : [{ label: 'map.dropdown.placeholder', value: 0 }]
     )
   )
-  selectedLinkIndex$ = new BehaviorSubject(0)
 
-  loading = false
-  error = null
+  private selectedLinkIndex$ = new BehaviorSubject(0)
+  private selectedStyleIndex$ = new BehaviorSubject(0)
 
-  selectedLink$ = combineLatest([
-    this.compatibleMapLinks$,
+  selectedSourceLink$ = combineLatest([
+    this.sourceLinks$,
     this.selectedLinkIndex$.pipe(distinctUntilChanged()),
-  ]).pipe(map(([links, index]) => links[index]))
+  ]).pipe(
+    map(([links, idx]) => links[idx]),
+    shareReplay(1)
+  )
+
+  styleLinks$ = this.selectedSourceLink$.pipe(
+    switchMap((src) => {
+      if (
+        src &&
+        src.type === 'service' &&
+        src.accessServiceProtocol === 'tms'
+      ) {
+        return from(
+          this.dataService.getGeodataLinksFromTms(
+            src as DatasetServiceDistribution,
+            false
+          )
+        ).pipe(
+          map(
+            (links) =>
+              links?.filter(
+                (l) =>
+                  l.type === 'service' &&
+                  l.accessServiceProtocol === 'maplibre-style'
+              ) || []
+          )
+        )
```

Comment:
Do we need this filter? Shouldn't all links returned by `getGeodataLinksFromTms` be `maplibre-style` links?

Reply:
Ok, got it, thanks for the explanation. Wondering if a small comment around this `map` would make sense to mention this case? In case you add one, could you also rename `l` to `link`?